rotkehlchen/api/rest.py

```diff
@@ -1323,14 +1323,31 @@ def user_premium_key_remove(self) -> Response:
         # else
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
-    def query_all_assets(self, filter_query: AssetsFilterQuery) -> Response:
-        """Returns all supported assets with pagination."""
+    def query_list_of_all_assets(
+            self,
+            filter_query: AssetsFilterQuery,
+            identifiers: Optional[List[str]],
+    ) -> Response:
+        """
+        If a set of identifiers is provided returns the information for those identifiers otherwise
+        returns all supported assets with pagination.
+        """
         assets, assets_found = GlobalDBHandler().retrieve_assets(filter_query=filter_query)
```

Comment:
hmmm what are you doing here? You are querying the entire list of assets from the DB and then filter it on the python side with a for loop? Come on man, the real solution here would be to add the identifier to the filter query as an optional argument.